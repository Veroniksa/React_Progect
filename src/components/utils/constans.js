export const AUTHORS = {
  HUMAN: "human",
  bot: "bot",
};

export const URL_PUBLIC = "https://api.spaceflightnewsapi.net/v3/articles";

export const REQUEST_STATUS = {
  IDLE: "idle",
  PENDING: "pending",
  SUCCESS: "success",
  FAILURE: "failure",
};

export const WEATHER_URL = "https://goweather.herokuapp.com/weather/Rome";

