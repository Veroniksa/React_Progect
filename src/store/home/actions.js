import { login } from "../../services/firebase";

export const HANDEL_LOGIN = "HOME::HANDEL_LOGIN";

export const handelLodin = (email, pass) => ({
  type: HANDEL_LOGIN,
  payload: {
    email,
    pass,
  },
});

export const handleLoginFb = (email, pass) => () => {
  try {
    login(email, pass);
    //setAuthed(true);
  } catch (e) {
    console.log(e);
    //setError()
  }
};
