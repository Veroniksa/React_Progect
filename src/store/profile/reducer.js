import { CHECK_BOX, TOOGLE_SHOW_NAME } from "./action";

const initialState = {
    showName: false,
    checkBox: false,
};

export const profileReducer = (state = initialState, action) => {
    switch(action.type) {
        case TOOGLE_SHOW_NAME: {
            return {
                ...state,
                showName: !state.showName
            };
        }
        case CHECK_BOX: {
            return {
                ...state,
                checkBox: !state.checkBox
            };
        }
        default:
            return state;
    } 
};