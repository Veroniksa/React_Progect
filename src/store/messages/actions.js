import { AUTHORS } from "../../components/utils/constans";

export const ADD_MESSAGE = "MESSAGESS::ADD_MESSAGE";
export const DELETE_MESSAGE = "MESSAGESS::DELETE_MESSAGE";

export const addMessage = (itemId, text, author) => ({
  type: ADD_MESSAGE,
  payload: {
    itemId,
    text,
    author,
  },
});

export const deleteMessage = (itemId, id) => ({
  type: DELETE_MESSAGE,
  payload: {
    itemId,
    id,
  },
});

let timer;
export const addMessageWithReplay = (itemId, text, author) => (dispatch) => {
  dispatch(addMessage(itemId, text, author));

  if (author === AUTHORS.HUMAN) {
    clearTimeout(timer);
    timer = setTimeout(() => {
      dispatch(addMessage(itemId, "Hello", AUTHORS.bot));
    }, 2000);
  }
};
