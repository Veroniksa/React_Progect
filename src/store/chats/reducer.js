import { ADD_CHAT, DELETE_CHAT } from "./actions";

const initalState = {
  items: [],
};

export const chatsReducer = (state = initalState, { type, payload }) => {
  switch (type) {
    case ADD_CHAT: {
      return {
        ...state,
        items: [...state.items, { id: `chat-${Date.now()}`, name: payload }],
      };
    }
    case DELETE_CHAT: {
      const newChats = state.items.filter(({ id }) => id !== payload);
      return {
        ...state,
        items: newChats,
      };
    }
    default:
      return state;
  }
};
