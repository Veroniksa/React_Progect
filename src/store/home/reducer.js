import { HANDEL_LOGIN } from "./actions";

const initialState = {
  email: [],
  pass: [],
};

export const homeReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case HANDEL_LOGIN: {
      return {
        ...state,
      };
    }
  }
};
