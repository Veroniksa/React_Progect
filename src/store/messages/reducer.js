import { DELETE_CHAT } from "../chats/actions";
import { ADD_MESSAGE, DELETE_MESSAGE } from "./actions";

const initialState = {
    messagesList: {},
};

export const messageReducer = (state = initialState, {type,payload}) => {
    switch(type) {
        case ADD_MESSAGE: {
            return {
                ...state,
                messagesList: {
                    ...state.messagesList,
                    [payload.itemId]: [
                        ...state.messagesList[payload.itemId] || [], 
                        {
                          id: `message-${Date.now()}`,
                          text: payload.text,
                          author: payload.author,
                        }
                    ]
                }
            };
        }
        case DELETE_MESSAGE: {
            const newChatMessages = state.messagesList[payload.itemId].filter(
                ({id}) => id !== payload.id
            );
            return {
                ...state,
                messagesList: {
                    ...state.messagesList,
                    [payload.itemId] : newChatMessages,
                }
            };
        }
        case DELETE_CHAT: {
            const newMessages = {...state.messagesList};
            delete state.messagesList[payload];
            
            return {
                ...state,
                messagesList: newMessages,
            };
        }
        default:
            return state;
    } 
};