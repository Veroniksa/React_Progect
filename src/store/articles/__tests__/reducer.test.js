import React from "react";
import { REQUEST_STATUS } from "../../../components/utils/constans";
import { GET_ARTICLES_PENDING } from "../actions";

import { newsReduser } from "../reducer";

describe("articles reducer", () => {
  it("hade case, returns state and request", () => {
    const expected = {
      state: null,
      request: {
        errror: null,
        status: REQUEST_STATUS.PENDING,
      },
    };

    expect(expected).toMatchSnapshot();
  });
});
