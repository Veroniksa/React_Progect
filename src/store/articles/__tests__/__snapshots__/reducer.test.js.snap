// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articles reducer hade case, returns state and request 1`] = `
Object {
  "request": Object {
    "errror": null,
    "status": "pending",
  },
  "state": null,
}
`;
